mindmeld/auto_annotator.py

```diff
@@ -835,5 +931,373 @@ def valid_entity_check(self, entity):
+
+        Args:
+            spans (list): List of tuples representing candidate spans (start_index, end_index + 1).
+        Returns:
+            selected_spans (list): List of the largest non-overlapping spans.
+        """
+        spans.sort(reverse=True)
+        selected_spans = []
+        for span in spans:
+            has_overlaps = [
+                span.has_overlap(selected_span) for selected_span in selected_spans
+            ]
+            if not any(has_overlaps):
+                selected_spans.append(span)
+        return selected_spans
+
+    @staticmethod
+    def _filter_duckling_candidates(candidates):
+        """Pipeline function to filter initial list of duckling candidates using heuristics.
+
+        Args:
+            candidates (list): List of duckling candidates
+        Returns:
+            filtered_candidates (list): List of filtered duckling candidates.
+        """
+        filtered_candidates = (
+            NoTranslationDucklingAnnotator._remove_unresolved_sys_amount_of_money(
+                candidates
+            )
+        )
+        return filtered_candidates
+
+    @staticmethod
+    def _remove_unresolved_sys_amount_of_money(candidates):
+        """Do not label candidate entities that are sys_amount-of-money but
+        do not have an "unknown" unit type.
+        """
+        return [
+            candidate
+            for candidate in candidates
+            if not (
+                candidate["dim"] == "amount-of-money"
+                and candidate["value"].get("unit") == "unknown"
+            )
+        ]
+
+
+class TranslationDucklingAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(
+        self,
+        app_path,
+        annotation_rules=None,
+        en_annotator=None,
+        translator=None,
+        language=None,
+        locale=None,
+        overwrite=False,
+        unannotate_supported_entities_only=True,
+        unannotation_rules=None,
+    ):
+        """Initializes a TranslationDucklingAnnotator.
+
+        Args:
+            app_path (str): The location of the MindMeld app.
+            annotation_rules (list): List of Annotation rules.
+            en_annotator (SpacyAnnotator): A Spacy Annotator with language set to English ("en").
+            translator (str): A translator to use such as 'GoogleTranslator' or 'NoOpTranslator'.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            overwrite (bool): Whether to overwrite existing annotations with conflicting spans.
+            unannotate_supported_entities_only (bool): Only allow removal of supported entities.
+            unannotation_rules (list): List of Annotation rules.
+        """
+        super().__init__(
```

Comment:
again, do the `*args and **kwargs` trick here to reduce verbosity. You can also pass in "extra" args that dont need to be used in `Annotator ` class so that you dont have to pop out args before calling super.